plugins/woo-ai/src/shared/productData.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Internal dependencies
+ */
+import { Attribute, ProductData, TinyContent } from './types';
+
+declare const tinymce: { get: ( str: string ) => TinyContent };
+
+const isElementVisible = ( element: HTMLElement ) =>
+	! ( window.getComputedStyle( element ).display === 'none' );
+
+const getCategories = (): string[] => {
+	const categoryCheckboxEls: NodeListOf< HTMLInputElement > =
+		document.querySelectorAll(
+			'#taxonomy-product_cat input[name="tax_input[product_cat][]"]:checked'
+		);
+
+	const tempCategories: string[] = [];
+
+	categoryCheckboxEls.forEach( ( el ) => {
+		if ( ! el.value.length ) {
+			return;
+		}
+
+		tempCategories.push( el.value );
+	} );
+
+	return tempCategories;
+};
+
+const getTags = (): string[] => {
+	const tagsEl: HTMLTextAreaElement | null = document.querySelector(
+		'textarea[name="tax_input[product_tag]"]'
+	);
+
+	const tags = tagsEl ? tagsEl.value.split( ',' ) : [];
+
+	return tags.filter( ( tag ) => tag !== '' );
+};
+
+const getAttributes = (): Attribute[] => {
+	const attributeSelectEls: NodeListOf< HTMLSelectElement > =
+		document.querySelectorAll(
```

Comment:
I believe this only selects any preexisting global attributes, and not any local attributes that have been added to the product. Is that the intention?

Reply:
Yeah, that's correct. Since the user needs to press "Save Attributes" to persist their attributes, I figured getting the current set from that input would be easier and not do any other DOM tricks to get the real-time value. The select value will be updated whenever the user clicks on Save Attributes, and that's enough for our use case IMO.